cmd/server/api/router.go

```diff
@@ -11,7 +12,12 @@ type Router interface {
 	ServeHTTP(w http.ResponseWriter, r *http.Request)
 }
 
-func Routes(h HandlerInterface, r Router) Router {
-	r.HandleFunc("/", h.PostRPCHandler).Methods("POST")
-	return r
+func Routes(h HandlerInterface, r Router, storage scan.Storage) Router {
+	router := r.(*mux.Router)
+	rpcProxyHandler := http.HandlerFunc(h.PostRPCProxyHandler)
+	erc721UniversalMintingHandler := http.HandlerFunc(h.UniversalMintingRPCHandler)
+
+	// Pass both handlers to the middleware and let it decide based on the JSON-RPC method and the contract address
+	router.Handle("/", PostRpcRequestMiddleware(rpcProxyHandler, erc721UniversalMintingHandler, storage)).Methods("POST")
```

Comment:
Is it possible to pass ` h handleInterface`  as an arg to the function `PostRpcRequestMiddleware`  and inside of it define (`rpcProxyHandler` and `erc721UnversalMintingHandler`)